### PR TITLE
FAI-663: Add initial support for remaining input features in CF as constrained

### PR DESCRIPTION
**Pull request by ruivieira**
[FAI-663](https://issues.redhat.com/browse/FAI-663)

Add counterfactuals support for additional features types as fixed/constrained.

> Many thanks for submitting your Pull Request :heart:! 
> 
> Please make sure that your PR meets the following requirements:
> 
> - [x] You have read the [contributors guide](https://github.com/kiegroup/kogito-runtimes#contributing-to-kogito)
> - [x] Pull Request title is properly formatted: `KOGITO-XYZ Subject`
> - [x] Pull Request title contains the target branch if not targeting main: `[0.9.x] KOGITO-XYZ Subject`
> - [x] Pull Request contains link to the JIRA issue
> - [x] Pull Request contains link to any dependent or related Pull Request
> - [x] Pull Request contains description of the issue
> - [x] Pull Request does not include fixes for issues other than the main ticket

<details>
<summary>
How to retest this PR or trigger a specific build:
</summary>

* <b>Pull Request</b>  
  Please add comment: <b>Jenkins retest this</b>
 
* <b>Quarkus LTS checks</b>  
  Please add comment: <b>Jenkins run LTS</b>

* <b>Native checks</b>  
  Please add comment: <b>Jenkins run native</b>

</details>

### CODE REVIEW

**Comment by ruivieira** [comment]
Jenkins retest this please

**Review by tteofili — APPROVED** [review]
LGTM

File: explainability/explainability-core/src/test/java/org/kie/kogito/explainability/local/counterfactual/entities/SimilarityTest.java [hunk 1]
```diff
@@ -127,6 +148,110 @@ void fixedBooleanSimilarity(int seed) {
         assertEquals(HIGHEST_SIMILARITY, x.similarity());
     }
 
+    @ParameterizedTest
+    @ValueSource(ints = { 0, 1, 2, 3, 4 })
+    void fixedBinarySimilarity(int seed) {
```
Comment (manstis) [inline explainability/explainability-core/src/test/java/org/kie/kogito/explainability/local/counterfactual/entities/SimilarityTest.java:153]: Just an observation... if you haves tests for _simularity_ you should/could have tests for _distance_ too.

**Review by manstis — APPROVED** [review]
LGTM. Minor comment about tests (but unrelated to this PR).

File: explainability/explainability-core/src/test/java/org/kie/kogito/explainability/local/counterfactual/entities/SimilarityTest.java (hunk 1 above)
Comment (ruivieira) [reply explainability/explainability-core/src/test/java/org/kie/kogito/explainability/local/counterfactual/entities/SimilarityTest.java:153]: Hi @manstis, thanks. Good point, I will add it in a separate PR if you agree.

**Pull request merged**